android/src/main/java/com/aerofs/reactnativeautoupdater/ReactNativeAutoUpdater.java

```diff
@@ -343,25 +345,38 @@ protected String doInBackground(String... params) {
                     jsCodeDir.mkdirs();
                 }
                 File jsCodeFile = new File(jsCodeDir, RNAU_STORED_JS_FILENAME);
+                File jsCodeFile = new File(jsCodeDir, RNAU_STORED_JS_TEMP_FILENAME);
                 output = new FileOutputStream(jsCodeFile);
 
-                byte data[] = new byte[4096];
+                byte data[] = new byte[12288];
                 int count;
-                while ((count = input.read(data)) != -1) {
-                    // allow canceling with back button
-                    if (isCancelled()) {
-                        input.close();
-                        return null;
+
+                try {
+                    while ((count = input.read(data)) != -1) {
+                        // allow canceling with back button
+                        if (isCancelled()) {
+                            input.close();
+                            return null;
+                        }
+                        output.write(data, 0, count);
                     }
-                    output.write(data, 0, count);
-                }
 
-                SharedPreferences prefs = context.getSharedPreferences(RNAU_SHARED_PREFERENCES, Context.MODE_PRIVATE);
-                SharedPreferences.Editor editor = prefs.edit();
-                editor.putString(RNAU_STORED_VERSION, params[1]);
-                editor.putLong(RNAU_LAST_UPDATE_TIMESTAMP, new Date().getTime());
-                editor.apply();
+                    // Rename temp js bundle file to main bundle file after download is completed so that crash does not occur.  Changes will be reflected in next app launch after the download
+                    File from = new File(jsCodeDir, RNAU_STORED_JS_TEMP_FILENAME);
+                    File to = new File(jsCodeDir, RNAU_STORED_JS_FILENAME);
+                    from.renameTo(to);
+
+                    SharedPreferences prefs = context.getSharedPreferences(RNAU_SHARED_PREFERENCES, Context.MODE_PRIVATE);
+                    SharedPreferences.Editor editor = prefs.edit();
+                    editor.putString(RNAU_STORED_VERSION, params[1]);
+                    editor.putLong(RNAU_LAST_UPDATE_TIMESTAMP, new Date().getTime());
+                    editor.apply();
+//                    Log.d("autoUpdate","Bundle Downloaded");
+                    } catch (Exception e){
+                        Log.e("autoUpdate", "exception ::: "+e);
```

Comment:
i have updated the changes. following error comes how to fix
package okhttp3 does not exist import okhttp3.OkHttpClient;

Reply:
you can [import lib in android](https://stackoverflow.com/questions/27435037/how-to-import-okhttp-library-to-android-studio#27435849) in app/build.gradle

Reply:
can you clearly give the line